packages/open-payments/src/client/grant.ts

```diff
@@ -0,0 +1,58 @@
+import { HttpMethod } from 'openapi'
+import { ClientDeps } from '.'
+import {
+  getASPath,
+  InteractiveGrant,
+  InteractiveGrantRequest,
+  NonInteractiveGrant,
+  NonInteractiveGrantRequest
+} from '../types'
+import { post } from './requests'
+
+interface RequestGrantArgs<T> {
+  url: string
+  request: T
+}
+
+export interface GrantRoutes {
+  requestInteractiveGrant(
+    args: RequestGrantArgs<InteractiveGrantRequest>
+  ): Promise<InteractiveGrant>
```

Comment:
My concern with splitting `requestInteractiveGrant`/`requestNonInteractiveGrant` is that I don't _think_ we've standardized what grant requests are interactive. And thus, a client might not be expected to know when to use either.
For example, Rafiki has made `incoming-payment`-only grants non-interactive, but I don't know that that's required:
- https://github.com/interledger/rafiki/pull/588

@sabineschaller or @njlie  or @adrianhopebailie might be able to correct me on this.

A compromise could be to do:
```suggestion
  requestInteractiveGrant(
    args: RequestGrantArgs<InteractiveGrantRequest>
  ): Promise<InteractiveGrant | NonInteractiveGrant>
```

Reply:
`requestNonInteractiveGrant` seems fine as is, since the request should just fail if the AS requires RO interaction for that grant.

Reply:
I separated these, since at some point, I thought the client need to know that they need to pass in `interact` ? 
https://github.com/interledger/open-payments/blob/main/openapi/auth-server.yaml#L93-L94

Reply:
Given our mix of grants for different resources needing interact or not, I've been assuming a client might just always assume a grant request requires interaction, but also be able to handle getting back a `NonInteractiveGrant` response.

Reply:
I'm still leaning towards
```suggestion
  requestInteractiveGrant(
    args: RequestGrantArgs<InteractiveGrantRequest>
  ): Promise<InteractiveGrant | NonInteractiveGrant>
```
and clients can utilize `isInteractiveGrant`/`isNonInteractiveGrant`.

I got the impression from Adrian that their policy agent AS might be determining what grant requests require interaction (vs having it Open Payments standardized).

Reply:
Ah, I see what you mean, it's more of a configurable, per-instance thing instead of standard, makes sense. 

Going to make changes to assume client will always require interaction, but call might return either or.

Reply:
I'm ok also keeping:
```
  requestNonInteractiveGrant(
    args: RequestGrantArgs<NonInteractiveGrantRequest>
  ): Promise<NonInteractiveGrant>
```
But maybe requiring the `interact` field to be specified isn't unreasonable? :thinking:

Reply:
Yeah even in the demo Postman example, there is an interact field present when requesting the incoming payment grant -> which returns a non-interactive one. 
We could run with a single call to make things simple for now